docs/tutorials/tabular_prediction/tabular-custom-model.md

```diff
@@ -0,0 +1,374 @@
+
+```{.python .input}
+from autogluon.features.generators import AutoMLPipelineFeatureGenerator
+
+feature_generator = AutoMLPipelineFeatureGenerator()
+X_clean = feature_generator.fit_transform(X)
+
+X_clean.head(5)
+```
+
+### Fit model
+
+We are now ready to fit the model with the cleaned features and labels.
+
+```{.python .input}
+custom_model = CustomRandomForestModel()
+# We could also specify hyperparameters to override defaults
+# custom_model = CustomRandomForestModel(hyperparameters={'max_depth': 10})
+custom_model.fit(X=X_clean, y=y_clean)  # Fit custom model
+
+# To save to disk and load the model, do the following:
+# load_path = custom_model.path
+# custom_model.save()
+# del custom_model
+# custom_model = CustomRandomForestModel.load(path=load_path)
+```
+
+### Predict with fit model
+
+Now that the model is fit, we can make predictions on new data. Remember that we need to perform the same data and label transformations to the new data as we did to the training data.
+
+```{.python .input}
+# Prepare test data
+X_test_clean = feature_generator.transform(X_test)
+y_test_clean = label_cleaner.transform(y_test)
+
+X_test.head(5)
+```
+
+Get raw predictions from the test data
+
+```{.python .input}
+y_pred = custom_model.predict(X_test_clean)
+print(y_pred[:5])
+```
+
+Note that these predictions are of the positive class (whichever class was inferred to 1). To get more interpretable results, do the following:
+
+```{.python .input}
+y_pred_orig = label_cleaner.inverse_transform(y_pred)
+y_pred_orig.head(5)
+```
+
+### Score with fit model
+
+By default, the model has an eval_metric specific to the problem_type. For binary classification, it uses accuracy.
+
+We can get the accuracy score of the model by doing the following:
+
+```{.python .input}
+score = custom_model.score(X_test_clean, y_test_clean)
+print(f'test score ({custom_model.eval_metric.name}) = {score}')
+```
+
+## Training a bagged custom model without TabularPredictor
+
+Some of the more advanced functionality in AutoGluon such as bagging can be done very easily to models once they inherit from AbstractModel.
+
+You can even bag your custom model in a couple lines of code. This is a quick way to get quality improvements on nearly any model:
+
+```{.python .input}
+from autogluon.core.models import BaggedEnsembleModel
+bagged_custom_model = BaggedEnsembleModel(CustomRandomForestModel())
+bagged_custom_model.fit(X=X_clean, y=y_clean, k_fold=10)  # Perform 10-fold bagging
+bagged_score = bagged_custom_model.score(X_test_clean, y_test_clean)
+print(f'test score ({bagged_custom_model.eval_metric.name}) = {bagged_score} (bagged)')
+print(f'Bagging increased model accuracy by {round(bagged_score - score, 4) * 100}%!')
+```
+
```

Comment:
might need to clarify how the data preprocessing works for bagged models. I recall there was a bug in one model at some point because it assumed the preprocessing was being re-run for each bagging fold.

Reply:
Added a bit of clarification on what happens inside the bagged model, but I don't think we need to mention the data preprocessing logic, as we currently have users work with `_preprocess` which works on a per-fold basis and is much safer.